lib/rexml/element.rb

```diff
@@ -621,17 +621,9 @@ def namespace(prefix=nil)
       end
       if prefix == ''
         prefix = "xmlns"
-      else
-        prefix = "xmlns:#{prefix}" unless prefix[0,5] == 'xmlns'
       end
-      ns = nil
-      target = self
-      while ns.nil? and target
-        ns = target.attributes[prefix]
-        target = target.parent
-      end
-      ns = '' if ns.nil? and prefix == 'xmlns'
-      return ns
+
+      namespaces[prefix].to_s
```

Comment:
It's changed to always returns a string, so the method comment needs to be updated
```diff
- #   b.namespace('nosuch') # => nil
+ #   b.namespace('nosuch') # => ""
```

Reply:
Oh, sorry.
This is an incompatible change, so I will change it to return nil.
Thanks!